File: algorithms/backtracking/subset-sum/csharp/SubsetSum.cs
public class SubsetSum
{
    public static int SubsetSumSolve(int[] arr, int target)
    {
        return Backtrack(arr, 0, target) ? 1 : 0;
    }

    private static bool Backtrack(int[] arr, int index, int remaining)
    {
        if (remaining == 0)
        {
            return true;
        }
        if (index >= arr.Length)
        {
            return false;
        }
        // Include arr[index]
        if (Backtrack(arr, index + 1, remaining - arr[index]))
        {
            return true;
        }
        // Exclude arr[index]
        if (Backtrack(arr, index + 1, remaining))
        {
            return true;
        }
        return false;
    }
}

File: algorithms/backtracking/subset-sum/typescript/subsetSum.ts
export function subsetSum(arr: number[], target: number): number {
    function backtrack(index: number, remaining: number): boolean {
        if (remaining === 0) {
            return true;
        }
        if (index >= arr.length) {
            return false;
        }
        // Include arr[index]
        if (backtrack(index + 1, remaining - arr[index])) {
            return true;
        }
        // Exclude arr[index]
        if (backtrack(index + 1, remaining)) {
            return true;
        }
        return false;
    }

    return backtrack(0, target) ? 1 : 0;
}
